rir/src/ir/CodeVerifier.cpp

```diff
@@ -234,6 +234,8 @@ void CodeVerifier::calculateAndVerifyStack(Code* code) {
     code->stackLength = max.ostack;
 }
 
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wunused-variable"
 void CodeVerifier::verifyFunctionLayout(SEXP sexp, InterpreterInstance* ctx) {
     assert(TYPEOF(sexp) == EXTERNALSXP and "Invalid SEXPTYPE");
```

Comment:
if the code verifier fails, the code is seriously broken and should not be executed. I think it would be a good idea to convert those asserts into Rf_error. The same goes for the pir verifier.